src/services/ledger/index.ts

```diff
@@ -617,6 +617,26 @@ export const LedgerService = (): ILedgerService => {
     }
   }
 
+  const getWalletIdByTransactionHash = async (
+    hash,
+  ): Promise<WalletId | CouldNotFindTransactionError | UnknownLedgerError> => {
```

Comment:
It looks like we've been keeping this more generic so far

```suggestion
  ): Promise<WalletId | LedgerError> => {
```